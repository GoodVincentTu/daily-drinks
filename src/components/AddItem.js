import React from 'react';

class AddItem extends React.Component {
  state = {
    title: '熟成紅茶',
    price: 20,
    notes: '微糖微冰',
  };

  handleChange = e => {
    const { name, type, value } = e.target;
    const val = type === 'number' ? parseFloat(value) : value;
    this.setState({ [name]: val });
  };

  handleClick = e => {
    e.preventDefault();
    const itemObj = {...this.state};
    this.props.handleAddItems(itemObj);
  }

  render() {
    return (
      <div>
        <fieldset>
          <label htmlFor="title">
            Name:
            <input 
              type="text"
              id="title"
              name="title"
              placeholder="Enter the name of the drink"
              required
              value={this.state.title}
              onChange={this.handleChange} />
          </label>

          <label htmlFor="price">
            Price
            <input 
              type="number"
              id="price"
              name="price"
              placeholder="Price"
              required
              value={this.state.price}
              onChange={this.handleChange} />
          </label>

          <label htmlFor="notes">
            Notes
            <textarea 
              name="notes"
              id="notes" 
              cols="30"
              rows="10"
              placeholder="Enter some additional notes"
              required
              value={this.state.notes}
              onChange={this.handleChange} />
          </label>

          <button
            onClick={this.handleClick}>
              Submit
          </button>
        </fieldset>
      </div>
    )
  }
}

export default AddItem;