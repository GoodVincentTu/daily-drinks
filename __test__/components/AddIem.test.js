import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AddItem from '../../src/components/AddItem';

describe('<AddItem/>', () => {
  it('renders', () => {
    shallow(<AddItem/>);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<AddItem/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  })
})