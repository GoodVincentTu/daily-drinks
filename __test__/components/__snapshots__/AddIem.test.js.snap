// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddItem/> matches the snapshot 1`] = `
<div>
  <fieldset>
    <label
      htmlFor="title"
    >
      Name:
      <input
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="Enter the name of the drink"
        required={true}
        type="text"
        value="熟成紅茶"
      />
    </label>
    <label
      htmlFor="price"
    >
      Price
      <input
        id="price"
        name="price"
        onChange={[Function]}
        placeholder="Price"
        required={true}
        type="number"
        value={20}
      />
    </label>
    <label
      htmlFor="notes"
    >
      Notes
      <textarea
        cols="30"
        id="notes"
        name="notes"
        onChange={[Function]}
        placeholder="Enter some additional notes"
        required={true}
        rows="10"
        value="微糖微冰"
      />
    </label>
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </fieldset>
</div>
`;
